src/sqlfluff/core/templaters/jinja.py

```diff
@@ -420,6 +433,35 @@ def _get_macros_path(self, config: FluffConfig) -> Optional[List[str]]:
                     return result
         return None
 
+    def _get_exclude_macros_path(self, config: FluffConfig) -> Optional[List[str]]:
+        """Get the list of macros paths to exclude from the provided config object.
+
+        This method searches for a config section specified by the
+        'exclude_macros_from_path' key. If the section is
+        found, it retrieves the value associated with that section and splits it into
+        a list of strings using a comma as the delimiter. The resulting list is
+        stripped of whitespace and empty strings and returned. If the section is not
+        found or the resulting list is empty, it returns None.
+
+        Args:
+            config (FluffConfig): The config object to search for the macros path
+                section.
+
+        Returns:
+            Optional[List[str]]: The list of macros paths if found, None otherwise.
+        """
+        if config:
+            exclude_macros_path = config.get_section(
+                (self.templater_selector, self.name, "exclude_macros_from_path")
+            )
+            if exclude_macros_path:
+                result = [
+                    s.strip() for s in exclude_macros_path.split(",") if s.strip()
+                ]
+                if result:
+                    return result
+        return None
```

Comment:
This method duplicates a lot of `_get_macros_path`. Looks like everything is the same apart from the key (either `load_macros_from_path` or `exclude_macros_from_path`). To avoid so much duplication, could you consolidate them?

Open to other options, but perhaps by making the key one of the arguments, and then making this whole method a more generic `@staticmethod`?

Reply:
Good callout - I have simplified these into one method that takes in a key argument.